examples/advanced/demo_optimizing_physics_parameter.py

```diff
@@ -0,0 +1,268 @@
+dtype = torch.float32
+
+# %%
+# Define the physics
+# ------------------
+#
+# In this first example, we use the convolution operator, defined in the :class:`deepinv.physics.Blur` class.
+# We also generate a random convolution kernel of motion blur
+
+generator = dinv.physics.generator.MotionBlurGenerator(
+    psf_size=(25, 25), rng=torch.Generator(device), device=device
+)
+true_kernel = generator.step(1, seed=123)["filter"]
+physics = dinv.physics.Blur(noise_model=dinv.physics.GaussianNoise(0.02), device=device)
+
+# %% Load and example image and compute the measurement
+x = dinv.utils.load_url_image(
+    dinv.utils.demo.get_image_url("celeba_example.jpg"),
+    img_size=256,
+    resize_mode="resize",
+).to(device)
+
+y = physics(x, filter=true_kernel)
+
+dinv.utils.plot([x, y, true_kernel], titles=["Sharp", "Blurry", "True kernel"])
+
+# %%
+# Define an optimization algorithm
+# --------------------------------
+#
+# The convolution kernel lives in the simplex: positive entries summing to 1.
+# We can use one of the most basic optimization algorithm -- Projected Gradient Descent.
+# The following function allows one to compute the gradient of the loss function with respect to the convolution kernel and the orthogonal projection onto the simplex, by a sorting algorithm (Reference: Large-scale Multiclass Support Vector Machine Training via Euclidean Projection onto the Simplex
+# Mathieu Blondel, Akinori Fujino, and Naonori Ueda)
+#
+
+
+def loss_fn(physics, x, y, filter):
+    y_hat = physics.A(x, filter=filter)
+    loss = (y_hat - y).pow(2).mean()
+    return 0.5 * loss
+
+
+def gradient(physics, x, y, filter):
+    filter = filter.clone().requires_grad_(True)
+    with torch.enable_grad():
+        y_hat = physics.A(x, filter=filter)
+        loss = torch.nn.functional.mse_loss(y_hat, y)
+        loss.backward()
+    return filter.grad
+
+
+@torch.no_grad
+def projection_simplex_sort(v):
+    shape = v.shape
+    B = shape[0]
+    v = v.view(B, -1)
+    n_features = v.size(1)
+    u = torch.sort(v, descending=True, dim=-1).values
+    cssv = torch.cumsum(u, dim=-1) - 1.0
+    ind = torch.arange(n_features, device=v.device)[None, :].expand(B, -1) + 1.0
+    cond = u - cssv / ind > 0
+    rho = ind[cond][-1]
+    theta = cssv[cond][-1] / rho
+    w = torch.maximum(v - theta, torch.zeros_like(v))
+    return w.reshape(shape)
+
+
+# Now we can define the projected gradient descent steps
+def projected_gradient_descent(physics, x, y, kernel_init, n_iter=100, stepsize=0.01):
+    kernel_hat = kernel_init.clone()
+    losses = []
+    for i in tqdm(range(n_iter)):
+        # gradient step
+        grad = gradient(physics, x, y, kernel_hat)
+        kernel_hat = kernel_hat - stepsize * grad
+        # projection step
+        kernel_hat = projection_simplex_sort(kernel_hat)
+        # loss
+        with torch.no_grad():
+            losses.append((physics.A(x, filter=kernel_hat) - y).pow(2).sum().item())
```

Comment:
if you move the gradient computation in the loop, you can change this line to avoid double computation
```suggestion
            losses.append(loss.item())
```